MD/md-30/README.md

```diff
@@ -0,0 +1,175 @@
+# MD-30: Movement Name Service
+
+- **Description**: Define the structure for the Movement Name Service (MNS) on the Movement blockchain, providing a user-friendly way for users to interact with names and rooms through bonding curve-based access keys.
+- **Authors**: [Primata](mailto:primata@movementlabs.xyz)
+- **Collaborators**: Community contributions welcome
+
+## Overview
+
+The Move Name Service (MNS) is a system that allows users to register human-readable names on the Movement blockchain. It is designed to facilitate user-friendly interactions, registrations, and transactions on-chain. Expanding beyond traditional name services, MNS introduces the concept of access keys tied to a bonding curve, which determines the minting and burning prices of these keys. The MNS also provides revenue generation mechanisms for both the protocol and name owners.
+
+MNS falls under the SoFi category, offering a decentralized, permissionless framework that encourages community participation to build and iterate on infrastructure and services centered on Names and Rooms.
+
+## Definitions
+
+- **Names**: Human-readable identifiers registered on-chain. Should be freely tradeable. Room ownership is tied to Name. Names domain is `.move`. E.g. `[owned_name].move`.
+- **Rooms**: Digital spaces associated with a registered Name, accessible by the Name's owner or key holders.
+- **Keys**: Access key to Name's rooms. Mintable and burnable according to a bonding curve.
+- **Bonding Curve**: A mathematical model used to determine minting and burning prices for access keys.
+
+## Desiderata
+
+### D1: Decentralized and Permissionless Name System
+
+**User Journey**: Users can register human-readable names on the blockchain and access related services.
+
+**Justification**: A decentralized name system is essential for empowering users with on-chain identities and ensuring they can interact with the blockchain without technical friction.
+
+**Guidance and suggestions**:
+
+- Names should be fully decentralized and governed by the community, adhering to the principles of openness and transparency.
+
+### D2: Bonding Curve for Key Minting and Burning
+
+**User Journey**: Users or services can mint or burn access keys based on a bonding curve.
```

Comment:
Maybe suggest some example bonding curves that have been used with Name Services. Are there any particular concerns we should have around the curve? Are there DOS vectors via economic attacks?

Reply:
From the get-go it's important to figure out what the price should be. This depends on what we expect regarding the token price, but the friend.tech bonding curve math is pretty good.